### ISSUE TITLE
Spira seems incompatible with activesupport >= 5.2.0

### ISSUE THREAD

**Issue by jperville** [ruby-rdf/spira#47]
Hello,

I tried to bundle update one of our existing projects which uses spira. Upgrading activesupport dependency from v5.1.6 to v5.2.0 seems to break spira, as shown in the stacktrace below:

```sh
bash-4.2$ bundle exec rspec ./spec/serializers/subject_spec.rb:41
Run options: include {:locations=>{"./spec/serializers/subject_spec.rb"=>[41]}}

SemanticEnrichment::Serializers::Subject
  #load
    returns a Subject (FAILED - 1)

Failures:

  1) SemanticEnrichment::Serializers::Subject#load returns a Subject
     Failure/Error: serializer.load(subject_use.repository, subject_use.hooks.first.to_s)
     NoMethodError:
       undefined method `clear' for nil:NilClass
     # ./.bundle/ruby/gems/spira-2.1.0/lib/spira/base.rb:288:in `reset_changes'
     # ./.bundle/ruby/gems/spira-2.1.0/lib/spira/base.rb:162:in `reload'
     # ./.bundle/ruby/gems/spira-2.1.0/lib/spira/base.rb:124:in `initialize'
     # ./.bundle/ruby/gems/spira-2.1.0/lib/spira/persistence.rb:195:in `new'
     # ./.bundle/ruby/gems/spira-2.1.0/lib/spira/persistence.rb:195:in `project'
     # ./.bundle/ruby/gems/spira-2.1.0/lib/spira/persistence.rb:176:in `for'
...
```

I traced the problem to [Spira::Base#reset_changes](https://github.com/ruby-rdf/spira/blob/develop/lib/spira/base.rb#L286-L289), which seems to assume too much about activesupport internals: for example, the `@changed_attributes` instance variable has been refactored out in activesupport v5.2.0 (it has been replaced with a method which returns a frozen HashWithIndifferentAccess, which cannot be reset anymore, thus breaking `Spira::Base#reset_changes`).

Can you investigate the issue? Meanwhile, I will add `gem 'activesupport', '>= 5.0', '< 5.2'` to my Gemfiles.

**Comment by abrisse** [comment]
We should use [ActiveModel::Dirty#clear_changes_information](https://github.com/rails/rails/blob/5-2-stable/activemodel/lib/active_model/dirty.rb#L195) instead which is the proper way to do it, and it exists in both ActiveModel 5.1 and 5.2.

I will try to provide a patch whenever I have some time on that overall dependency constraint.

**Comment by abrisse** [comment]
Fixed in e75e409a1d660c8bcb8d1af39498bca4b405f88e. Release new version 3.1.0
